webextension/scripts/settings.js

```diff
@@ -15,6 +15,12 @@ $(function() {
   $('#view-setting').click(switchTabWindow)
   $('#view-setting').children('input,label').click((e) => { e.stopPropagation() })
   $('input[type=checkbox],input[type=radio]').change(saveOptions)
+  $('#auto-archive-age').change((e) => {
+    $('#auto-archive-setting').prop('checked', true)
```

Comment:
$('#auto-archive-setting').addClass('selected-prior')
This addition would fix the issue of private mode not clearing check-box.